Thinkful: Unit 3 - Drill - Describing Data


    Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?
    Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?
    Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?
    Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [21]:
import pandas as pd
import numpy as np
import statistics

In [17]:
df = pd.DataFrame()
df['name'] = ['Greg','Marcia','Peter','Jan','Bobby','Cindy','Oliver']
df['age'] = [14, 12, 11, 10, 8, 6, 8]
df['age']

0    14
1    12
2    11
3    10
4     8
5     6
6     8
Name: age, dtype: int64

In [30]:
print('The mean age of the Braddy Bunch is', (np.mean(df['age'])))
print('The median age of the Braddy Bunch is', (np.median(df['age'])))
print('The mode age of the Braddy Bunch is', (statistics.mode(df['age'])))
print('The variance for ages of the Braddy Bunch is', (np.var(df.age)))
print('The standard deviation of the ages of the Braddy Bunch is', (np.std(df['age'])))
print('The standard error for the ages of the Braddy Bunch is', (np.std(df['age'], ddof=1) / np.sqrt(len(df['age']))))


The mean age of the Braddy Bunch is 9.857142857142858
The median age of the Braddy Bunch is 10.0
The mode age of the Braddy Bunch is 8
The variance for ages of the Braddy Bunch is 6.408163265306122
The standard deviation of the ages of the Braddy Bunch is 2.531435020952764
The standard error for the ages of the Braddy Bunch is 1.0334540197243192


I would choose the mean and standard deviation to describe these data because the mean gives us the most precise way of thinking about how old these group of children were, while the standard deviation helps show how close in age they were.

Cindy has a birthday

In [35]:
df.loc[df.name == 'Cindy', 'age'] = 7

In [37]:
print('The mean age of the Braddy Bunch is', (np.mean(df['age'])))
print('The median age of the Braddy Bunch is', (np.median(df['age'])))
print('The mode age of the Braddy Bunch is', (statistics.mode(df['age'])))
print('The variance for ages of the Braddy Bunch is', (np.var(df.age)))
print('The standard deviation of the ages of the Braddy Bunch is', (np.std(df['age'])))
print('The standard error for the ages of the Braddy Bunch is', (np.std(df['age'], ddof=1) / np.sqrt(len(df['age']))))

The mean age of the Braddy Bunch is 10.0
The median age of the Braddy Bunch is 10.0
The mode age of the Braddy Bunch is 8
The variance for ages of the Braddy Bunch is 5.428571428571429
The standard deviation of the ages of the Braddy Bunch is 2.32992949004287
The standard error for the ages of the Braddy Bunch is 0.9511897312113418


The median change as well as all of the measures of variance

Replace Oliver with 1-year-old Jessica

In [39]:
df.loc[df.name == 'Oliver', ['name', 'age']] = 'Jessica', 1

In [40]:
df

name  age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    7
6  Jessica    1

In [42]:
print('The mean age of the Braddy Bunch is', (np.mean(df['age'])))
print('The median age of the Braddy Bunch is', (np.median(df['age'])))
print('The mode age of the Braddy Bunch is', ', there is no mode')
print('The variance for ages of the Braddy Bunch is', (np.var(df.age)))
print('The standard deviation of the ages of the Braddy Bunch is', (np.std(df['age'])))
print('The standard error for the ages of the Braddy Bunch is', (np.std(df['age'], ddof=1) / np.sqrt(len(df['age']))))

The mean age of the Braddy Bunch is 9.0
The median age of the Braddy Bunch is 10.0
The mode age of the Braddy Bunch is , all ages are the mode
The variance for ages of the Braddy Bunch is 15.428571428571429
The standard deviation of the ages of the Braddy Bunch is 3.927922024247863
The standard error for the ages of the Braddy Bunch is 1.6035674514745462


I still like the mean and the standard deviation as descriptors of these data.

 On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

    Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [43]:
df2 = pd.DataFrame()
df2['magazine'] = ['TV_Guide', 'Entertainment_Weekly', 'Pop_Culture_Today', 'SciPhi_Phanatic']
df2['Percent_Fans'] = [0.2, 0.23, 0.17, 0.05]
df2

magazine  Percent_Fans
0              TV_Guide          0.20
1  Entertainment_Weekly          0.23
2     Pop_Culture_Today          0.17
3       SciPhi_Phanatic          0.05

In [45]:
np.mean(df2.Percent_Fans)

0.16250000000000003

I would estimate that 16.25% of Americans were fans of the show on its 50th anniversary. But of course we would need to know the total number of readers for each magazine to actually estimate this as knowing the population size of each magazine will allow us to properly compare them.